packages/wallet/src/redux/protocols/advance-channel/reducer.ts

```diff
@@ -315,3 +311,39 @@ function channelAdvanced(channel: ChannelState, commitmentType: CommitmentType):
     lastCommitment.commitmentType > commitmentType
   );
 }
+
+function nextCommitment(channel: ChannelState, commitmentType: CommitmentType): Commitment {
+  const lastCommitment = getLastCommitment(channel);
+  const penultimateCommitment =
+    lastCommitment.turnNum > 0 ? getPenultimateCommitment(channel) : undefined;
+  if (commitmentType === CommitmentType.Conclude) {
+    if (!penultimateCommitment) {
+      throw new Error('Attempted to conclude a channel that only contains a setup commitment');
+    }
+    return nextConcludeCommitment(lastCommitment, penultimateCommitment);
+  } else {
+    const next = nextSetupCommitment(lastCommitment);
+    if (next === 'NotASetupCommitment') {
+      throw new Error('lastCommitment was not a setup commitment');
+    }
+    return next;
+  }
+}
+
+// This only works for 2 or 3 players
```

Comment:
Maybe throw an error if the parameters of the channel don't match this requirement, rather than having this comment?